packages/core/src/codewhisperer/region/regionProfileManager.ts

```diff
@@ -367,4 +450,63 @@ export class RegionProfileManager {
 
         return c
     }
+
+    // Should reset cache when users signout in case users want to change to a different connection
+    async resetCache() {
+        await globals.globalState.update('aws.amazonq.regionProfiles.cachedResult', {
+            isAcquired: false,
+            result: undefined,
+        })
+    }
+
+    private async tryAcquireLock(): Promise<CachedApiResultWithLock | undefined> {
```

Comment:
This is rather delicate logic. And it is straightforward to generalize it. Why is it buried in a random module?

Note that https://github.com/aws/aws-toolkit-vscode/blob/70ba83fc066cfe8a1f9f50e5eb73a5e1b51d42f4/packages/core/src/shared/crashMonitoring.ts#L432 attempted to achieve something similiar, except using the filesystem. Though it has some known issues.

Reply:
btw Justin, i'm going to use #7151 to replace this pr for this purpose (if everything works just fine there), because the diff are pretty large so would be easier to just have another PR.